### PR TITLE
Migrate Blog & Blog Post page to ES6

### LINKED ISSUE #613: Migrate React Components to ES6 syntax
React 15.5 signals the imminent deprecation of the `React.createClass` method (moving it to a separate component). In the interest of staying up to date, it would be great if we can update our components to use the ES6 syntax. This doesn't have to happen all at once (in fact, it would be easier to test if it doesn't), but if anyone wants to help out, they can update a component and open a PR. Just drop a comment in this issue before starting so you don't end up duplicating effort (don't wait for someone to tell you to go ahead though, just start).

[This post](https://toddmotto.com/react-create-class-versus-component/) seems to give a good overview of the differences between the syntaxes.

### PR DESCRIPTION
**Pull request by akshatnitd**
Fixes #613 

It migrates the blog and blog-post page to es6

### CODE REVIEW

File: app/react/pages/blog.jsx [hunk 1]
```diff
@@ -5,12 +5,26 @@ import DataCard from "../components/data-card/data-card.jsx";
...
-export default React.createClass({
-  componentWillMount: function () {
+export default class BlogList extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      posts: [],
+      pagesLoaded: 0,
+      categories: [
```
Comment (gvn) [inline app/react/pages/blog.jsx]: Collapse this into 1 line: `categories: [],`

File: app/react/pages/blog-post/blog-post.jsx [hunk 2]
```diff
@@ -2,18 +2,22 @@ import React from "react";
 import Moment from "moment-timezone";
 import Service from "../../../js/backend.js";
 
-export default React.createClass({
-  componentWillMount: function () {
+export default class BlogPost extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (gvn) [inline app/react/pages/blog-post/blog-post.jsx]: Add empty line above for readability.

**Comment by gvn** [comment]
I like how this approach precludes the need for method binding in the constructor. @alanmoo we should consider doing this on Network as well. 👍

**Comment by akshatnitd** [comment]
@gvn Made the requested changes :) Will make changes in the other components using the same approach :+1:

**Comment by gvn** [comment]
Thanks @akshatnitd !

**Pull request merged**